services/gateway-api/src/circuit_breaker/health_monitor.py

```
@@ -0,0 +1,450 @@
+            name: Service name
+            health_check: Async function that returns bool (True = healthy)
+            config: Health check configuration
+        """
+        self._services[name] = ServiceHealthMetrics(
+            service_name=name,
+            status=HealthStatus.UNKNOWN
+        )
+        self._health_checks[name] = health_check
+        self._configs[name] = config or HealthCheckConfig()
+
+        logger.info(f"Registered health monitoring for service: {name}")
```

Ruff (0.14.8)

Logging statement uses f-string

(G004)